samples/minimum_criteria.rst

```diff
@@ -0,0 +1,82 @@
+  * Should not use zasserts or asserts in samples.
+
+2. We shall not use the sample as a testcase.
++++++++++++++++++++++++++++++++++++++++++++++
+  * A feature should already have testcases under the tests category before the sample gets added. Testcases are mandatory, and having samples may be optional.
+  * The primary purpose of a sample is to provide a reference to the user.
+
+3. Should ideally run on multiple platforms and architectures.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize use of platform_allow and architecture filters as it would limit the testing on the mentioned platforms and architectures. Reference: `Testcase Docs`_
+  * Make use of integration_platforms to limit the scope when there are timing or resource constraints.
+  * Make the sample as generic as possible. Avoid making a sample platform specific unless it is for particular hardware.
+
+  .. _Testcase Docs: https://docs.zephyrproject.org/latest/develop/test/twister.html?highlight=test%20runner#test-cases.
+
+4. Twister should be able to run every sample, i.e., every sample must have a YAML file.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  **For example:**
+
+  .. code-block:: console
+
+    tests:
+      sample.kernel.cond_var:
+        integration_platforms:
+          - native_posix
+        tags: kernel condition_variables
+        harness: console
+        harness_config:
+          type: one_line
+          regex:
+            - ".*Waited and joined with 3 threads"
+
+  **Guidelines for Samples:**
+    * Minimize the use of platform_allow and architecture filters.
+    * Do not mark the test as build only unless necessary.
+    * Any test case-specific configuration options are added using extra_args or extra_configs options in the YAML file. The prj.conf file should have the in-common configuration options.
+    * There can be multiple testcases in the sample YAML to verify if the feature works with different configurations. For example, different logging modes are tested using multiple test scenarios in samples/subsys/logging/syst.
+    * Sample output can be validated leveraging the harness_config regex option, wherever applicable.
+
+5. A sample should provide sufficient coverage of a subsystem, feature, or module.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  **DO's:**
+    * Samples should provide integration testing.
+    * Cover the most common and important use cases of the functionality.
+    * Keep the code simple and easy to read. Example: samples/philosophers.
+
+  **DONT's:**
+    * Samples should not be testing the negative or edge case behaviors.
+    * Should not be unit tests.
+    * Should not incorporate more than one subsystem or feature.
+
+6. Must have a README clearly explaining the purpose of the sample and its HW requirements and giving sample output, if applicable.
```

Comment:
This header (and others) are way too long. Can you please shorten them, here for example it should be:

"Samples must be documented"

And then add the test as part of the text below or use it to lead to the text in the bullets.